interview-questions.md

```diff
@@ -8,36 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: If this happens you can run a new migration to add the foreign(this spelling tortured me this whole explanation) key to the studen model. Using something like "rails generate migration AddCohortRefToStudents cohort:references"
+
 
-Researched answer:
+Researched answer: Ok. 😤 So this was hard to research and I couldn't really nail down 1 source for my answer. So the following is a combination of different sets of information that I was able to pull from various sources. First I found a bunch of stuff about "ALTER TABLE" but I assumed that we were talking about how to fix this in rails so that didn't work(hence my anser above). So i went to the rails resource from the syllibus which went over how to add a column to a table cuz I thought it would be as easy as finding it there.(it wasn't) but that is where i found "add_reference". So I googled that and that led me to what adding a thhe active record website that had a good explanation of "add_reference" but it also showed me that there was an option for add_foreign key😳 which seems like it would have been more straight forward 😤
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Create and Update must always be passed. Create because without it you have nothing. It is used to create a new resource. Update because it doe exactly what it's name says. It is used to update the information without it the server wouldn't know what data to us to update the resource you created using Create(BOOM!🤯 full circle)
+
+Researched answer: Soooo again no where I could find just straight up answers the question. I even did a ⌘ + f in the syllabus for 'Create'(it appears 21 times in the params lesson btw but only 5 times in the RESTful routes lesson). Mostly I found a bunch of explanations about what create does. Which is creates your resource. Not a lot of explanation as to why you have to pass it though. I think most people felt it was self explanitory. But yes you have to pass create because that is what creates your resource. You can not pass it if you just don't want to work cuz you will have nothing to work on.
```

Comment:
Yep this is a weird question because what exactly does the question even mean when it asks which RESTful routes...right? But in the context of the LEARN classroom, we can go to the syllabus into the Rails RESTful routes and CRUD actions page lesson. Near the top we'll find a picture of the RESTful routes. Since RESTful routes are a set of conventions for defining URLs and HTTP methods used to access resources in the rails app, I would assume this question is asking which of those routes in that picture include params in the path, which would be show, edit, update, destroy.